naga/src/diagnostic_filter.rs

```diff
@@ -59,22 +59,35 @@ impl Severity {
 #[cfg_attr(feature = "deserialize", derive(Deserialize))]
 #[cfg_attr(feature = "arbitrary", derive(Arbitrary))]
 pub enum FilterableTriggeringRule {
+    Standard(StandardFilterableTriggeringRule),
+    Unknown(Box<str>),
+    User([Box<str>; 2]),
```

Comment:
Just a thought: `[Box<str>; 2]` makes `FilterableTriggeringRule` five words long, since `Box<str>` is a fat pointer, and the enum discriminant ends up being expanded to a full word to preserve alignment. But using `Box<String>` and `Box<[String; 2]>` would let `FilterableTriggeringRule` remain two words long, and these are probably easier to work with.

The `Unknown` and `User` cases are ideally going to be rare: anything that does become widespread we'd add support for, with a proper enum variant of its own. So the impact of the double indirection should be low.

Reply:
Agreed on indirection to minimize stack usage of the common case being worth it. I feel silly I hadn't thought to push the two `User` tokens behind a `Box`, honestly.

Reply:
Fixed with a force-push to 100b63d4.